packages/skeleton-svelte/src/lib/components/Rating/Rating.svelte

```diff
@@ -0,0 +1,125 @@
+<script lang="ts">
+	import type { RatingProps } from './types.js';
+
+	let {
+		value = $bindable(0),
+		max = 5,
+		interactive = false,
+		fraction = 1,
+		// Root
+		base = 'flex w-full',
+		text = 'text-token',
+		fill = 'fill-token',
+		justify = 'justify-center',
+		spaceX = 'space-x-2',
+		classes = '',
+		// Item ---
+		itemBase = 'w-full',
+		itemAspect = 'aspect-[1/1]',
+		itemClasses = '',
+		// Snippets ---
+		emptyIcon,
+		fullIcon
+	}: RatingProps = $props();
+
+	let hoverValue = $state(0);
+	let hovering = $state(false);
+	let lastSelectedFraction = $state(1);
+
+	function onRatingClick(event: Event, order: number) {
+		// handling mouse
+		if (event instanceof PointerEvent) {
+			const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+			console.log(event);
+			const fractionWidth = ratingRect.width / fraction;
+			const left = event.clientX - ratingRect.left;
+			lastSelectedFraction = Math.floor(left / fractionWidth) + 1;
+		}
+		// handling keyboard
+		else {
+		}
+		value = order + lastSelectedFraction / fraction;
+	}
+
+	function onRatingHover(event: Event, order: number) {
+		// handling mouse
+		if (event instanceof MouseEvent) {
+			const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+			const fractionWidth = ratingRect.width / fraction;
+			const left = event.clientX - ratingRect.left;
+			lastSelectedFraction = Math.floor(left / fractionWidth) + 1;
+		}
+		// handling keyboard
+		else {
+		}
+		hovering = true;
+		hoverValue = order + lastSelectedFraction / fraction;
+	}
+
+	function onRatingLeave() {
+		hovering = false;
+	}
+</script>
+
+{#snippet rating(order, percentage)}
+	{#if interactive}
+		<button
+			type="button"
+			class="relative h-full w-full"
+			onclick={(event) => onRatingClick(event, order)}
+			onmousemove={(event) => onRatingHover(event, order)}
+			onmousedown={(event) => onRatingHover(event, order)}
+			onfocus={(event) => onRatingHover(event, order)}
+		>
+			<div class="clip-left absolute left-0 top-0 flex h-full w-full items-center justify-center" style="--clip_value: {percentage * 100}%">
```

Comment:
There's a ton of inline classes that should likely be abstracted to `baseX` style props.

Reply:
I don't think it makes sense to abstract those classes, they are crucial for the functionality of the component and should not be changed by the user IMHO.
And I am not sure what to call the element if abstracted, maybe you have a suggestion...

Reply:
The problem her is we can't promote the library as having faux-headless support if some classes are hardcoded. Even if they're structurally important, consider moving them to a `base` prop of some sort.